### PR TITLE
PODS-4161 Updates to allow PSPs into the service

### PR DESCRIPTION
**Pull request by madhuranakate**

### CODE REVIEW

File: app/controllers/actions/AuthAction.scala [hunk 1]
```diff
@@ -19,21 +19,25 @@ package controllers.actions
...
+import uk.gov.hmrc.domain.{PsaId, PspId}
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.HeaderCarrierConverter
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class AuthActionImpl @Inject()(override val authConnector: AuthConnector, config: FrontendAppConfig,
-                               val parser: BodyParsers.Default)
-                              (implicit val executionContext: ExecutionContext) extends AuthAction with
+class AuthImpl @Inject()(override val authConnector: AuthConnector,
```
Comment (grantrobson) [inline app/controllers/actions/AuthAction.scala:36]: Do we need the Inject() here? I don't think these values are injected, they are passed in explicitly by object creation in same file

File: app/controllers/actions/AuthAction.scala [hunk 2]
```diff
@@ -59,19 +63,44 @@ class AuthActionImpl @Inject()(override val authConnector: AuthConnector, config
...
+      block(AuthenticatedRequest(request, id, None, Some(PspId(getPspId(enrolments)))))
+    }
+
+  private def getPsaId(enrolments: Enrolments): String =
     enrolments.getEnrolment("HMRC-PODS-ORG").flatMap(_.getIdentifier("PSAID")).map(_.value)
-      .getOrElse(throw new PsaIdNotFound)
+      .getOrElse(throw new IdNotFound)
+
+  private def getPspId(enrolments: Enrolments): String =
+    enrolments.getEnrolment("HMRC-PODS-ORG").flatMap(_.getIdentifier("PSPID")).map(_.value)
```
Comment (grantrobson) [inline app/controllers/actions/AuthAction.scala:84]: Shouldn't this be HMRC-PODSPP-ORG? Or is this just temporary until the enrolments ticket is done?

**Review by vinnicombe — APPROVED** [review]

File: app/controllers/actions/AuthAction.scala (hunk 1 above)
Comment (madhuranakate) [reply app/controllers/actions/AuthAction.scala]: Updated

File: app/controllers/actions/AuthAction.scala (hunk 2 above)
Comment (madhuranakate) [reply app/controllers/actions/AuthAction.scala:84]: This needs to be consistent with manage

**Pull request merged**